src/main/java/com/gluonhq/substrate/util/FileOps.java

```diff
@@ -147,4 +150,46 @@ public static Path copyFile(Path source, Path destination)  {
         }
         return destination;
     }
+
+    public static void deleteDirectory(Path start) throws IOException {
+        Files.walkFileTree(start, new HashSet(), Integer.MAX_VALUE, new FileVisitor<>() {
+            @Override
+            public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                Files.delete(file);
+                return FileVisitResult.CONTINUE;
+
+            }
+
+            @Override
+            public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+                return FileVisitResult.TERMINATE;
+
+            }
+
+            @Override
+            public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+                Files.delete(dir);
+                return FileVisitResult.CONTINUE;
+            }
+        });
+    }
+
+    public static void copyDirectory(Path source, Path destination) {
+        copyFile(source, destination);
+        if (Files.isDirectory(source)) {
+            try {
+                Files.list(source)
+                        .map(Path::getFileName)
+                        .forEach(fileName ->
+                            copyDirectory(source.resolve(fileName), destination.resolve(fileName)));
+            } catch (IOException e) {
+                Logger.logSevere("Error copying " + source.toString() + ": " + e.getMessage());
```

Comment:
You hide the exception here. The caller is not notified and will assume all is fine.

Reply:
I know, but there is a recursive method, and the exception within the forEach can't be propagated. I'll use a regular for loop, unless there is a better way